foreman_stub.py

```diff
@@ -9,61 +9,146 @@
 from collections import defaultdict
 from urllib.parse import urlparse
 
-from flask import Flask, jsonify, request
+from flask import Flask, abort, jsonify, request
 
 
 DEBUG = os.getenv('DEBUG', 'false').lower() not in ('false', '0', '', 'off')
 
 if DEBUG:
-    import pip
-    pip.main(['install', 'q', 'epdb'])
+    try:
+        import epdb
+    except ImportError:
+        pass
+    else:
+        epdb.serve()
 
-app = Flask(__name__)
+app = Flask(__name__)  # pylint: disable=invalid-name
 
 PAGECACHE = defaultdict(dict)
+REPORTS = []
 
 
 @app.before_first_request
 def build_pagecache():
     """Read JSON fixtures into in-memory cache."""
     jfiles = glob.glob('fixtures/*.json')
     for jfile in jfiles:
-        with open(jfile, 'r') as f:
-            jdata = json.load(f)
+        with open(jfile, 'r') as fixture:
+            jdata = json.load(fixture)
         key = urlparse(jdata['url']).path[7:]
         subkey = jdata['page']
         PAGECACHE[key][subkey] = jdata['data']
 
+        if 'id' in jdata['data'] and key == f"/hosts/{jdata['data']['id']}":
+            PAGECACHE[f"/hosts/{jdata['data']['name']}"][subkey] = jdata['data']
+
 
 def get_page_num():
-    """Return page number if present."""
-    pagenum = request.args.get('page')
-    if pagenum:
-        pagenum = int(pagenum)
-    return pagenum
+    """Return page number."""
+    return int(request.args.get('page', 1))
```

Comment:
Wouldn't it crash on an empty string?

Reply:
You're right, but you just shouldn't do that ;) I'll modify this.

Reply:
I think that I want only related changes in this PR. It'll shrink it to the reviewable size. Such things are just a distraction.

Reply:
Actually, the default value of 1 change is needed for the host detail page to work correctly.

Reply:
well, then maybe
```suggestion
    return int(request.args.get('page', 1) or 1)
```

Reply:
I was thinking about that, but `return int(request.args.get('page') or 1)` is probably sufficient.